src/components/register/register.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { ReactElement, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { IRegisterFormProperties } from '../../@types/types';
+import './register.scss';
+
+export function Register(): ReactElement {
+
+  const history = useHistory();
+
+  const [form, setState] = useState<IRegisterFormProperties>({
+    username: '',
+    password: '',
+    email: '',
+  });
+
+  function redirectToLogin() {
+    history.push('/login');
+  }
```

Comment:
The above function seems like a waste. It doesn't really bring anything to the table, and surely should be either out of the component so it's reusable, or simply not exist.

Did you consider something along the lines of the following approach?

```tsx
<a href="/register" onClick={() => history.push('/register')}>register</a>
```